File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/IceSkates.java
package org.rexcellentgames.burningknight.entity.item.accessory.equippable;

public class IceSkates extends Equippable {
	@Override
	public void onEquip(boolean load) {
		super.onEquip(load);
		this.owner.iceResitant += 1;
	}

	@Override
	public void onUnequip(boolean load) {
		super.onUnequip(load);
		this.owner.iceResitant -= 1;
	}
}

File: core/src/org/rexcellentgames/burningknight/entity/item/accessory/equippable/IceSkates.kt
package org.rexcellentgames.burningknight.entity.item.accessory.equippable

class IceSkates : Equippable() {
	override fun onEquip(load: Boolean) {
		super.onEquip(load)
		this.owner.iceResitant += 1
	}

	override fun onUnequip(load: Boolean) {
		super.onUnequip(load)
		this.owner.iceResitant -= 1
	}

	override fun canBeUpgraded(): Boolean {
		return false
	}

	override fun canBeDegraded(): Boolean {
		return false
	}
}